src/coreclr/src/pal/src/memory/pressure.cpp

```diff
@@ -0,0 +1,177 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "pal/palinternal.h"
+#include "pal_assert.h"
+#include "pal.h"
+
+#include <poll.h>
+#include <errno.h>
+
+static HANDLE pollingThread;
+static HANDLE notificationHandles[2];
+
+static int CreatePSIMemTrigger(const char *type, int stallAmountMicro, int timeWindowMicro)
+{
+    int fd = open("/proc/pressure/memory", O_RDWR | O_NONBLOCK | O_CLOEXEC);
```

Comment:
I am trying to play with this, and the permissions on `/proc/pressure/memory` prevent anyone who is non-root from writing to it at all:

```
$ ls -l /proc/pressure/
total 0
-r--r--r--. 1 root root 0 May 15 11:49 cpu
-r--r--r--. 1 root root 0 May 15 11:49 io
-r--r--r--. 1 root root 0 May 15 11:49 memory
```

Did you have this working on some Linux distro with some kernel version? I took a look at kernel code:

https://github.com/torvalds/linux/blob/1ae7efb388540adc1653a51a3bc3b2c9cef5ec1a/kernel/sched/psi.c#L1333-L1341 and it creates these `/proc/pressure/` devices with `proc_create(..., 0, ....)` call. That `0` means, as far as I can tell, that it's read-only for all users. A `sudo` bypasses the this-file-is-read-only-check.